multipatch_analysis/ui/acq4_modules.py

```diff
@@ -56,3 +71,266 @@ def load_from_dm_clicked(self):
         
     def window(self):
         return self.ui
+
+
+class MultiPatchMosaicEditorExtension(QtGui.QWidget):
+    def __init__(self, mosaic_editor):
+        self.mosaic_editor = mosaic_editor
+
+        QtGui.QWidget.__init__(self)
+        self.layout = QtGui.QGridLayout()
+        self.setLayout(self.layout)
+
+        self.load_btn = QtGui.QPushButton("Load 20x")
+        self.layout.addWidget(self.load_btn, 0, 0)
+        self.load_btn.clicked.connect(self.load_clicked)
+
+        self.create_json_btn = QtGui.QPushButton("Save Cell Location Json")
+        self.layout.addWidget(self.create_json_btn, 1, 0)
+        self.create_json_btn.clicked.connect(self.create_lims_json)
+
+        self.create_test_btn = QtGui.QPushButton('Testing Button')
+        self.layout.addWidget(self.create_test_btn, 2, 0)
+        self.create_test_btn.clicked.connect(self.test_button)
+
+    def test_button(self):
+        """place to test things
+        """
+        items = self.mosaic_editor.canvas.items 
+        markers = [i for i in items if isinstance(i, MarkersCanvasItem)]
+        if len(markers) != 1:
+            raise Exception("Must have exactly 1 Markers item in the canvas.")
+        pipettes = markers[0].saveState()['markers']
+        for cell in pipettes:
+            p = cell[1]
+            for i in items:
+                try:
+                    if i.opts['handle'].name() == self.image_20:
+                        image = i
+                except AttributeError:
+                    pass
+            print(markers[0].graphicsItem().mapToItem(image.graphicsItem(),pg.Point(*p[:2])).x())
+            print(markers[0].graphicsItem().mapToItem(image.graphicsItem(),pg.Point(*p[:2])).y())
+
+        
+
+    
+
+    def load_clicked(self):
+        """
+        Checks that directory is pointed at a slice
+        Check LIMS for 20x biocytin image and if 20x image is saved locally.
+        If image not saved locally then copies file from LIMS.
+        Opens button to create a JSON of the cell locations
+        """
+        self.base_dir = self.mosaic_editor.ui.fileLoader.baseDir()
+        if self.base_dir.info()['dirType'] == 'Slice':
```

Comment:
minor note: it's nice to avoid these really long if-blocks. In this case, you could instead write:
```
if self.base_dir.info()['dirType'] != 'Slice':
    raise Exception('No Slice Selected')
```
And then de-indent the rest of the block

Reply:
ok